### ISSUE TITLE
Docker deploy command requires filename without extension

### ISSUE THREAD

**Issue by arun-gupta** [docker/docker#25855]
Deploying a DAB as `docker deploy hellojavaee.dab` gives the error:

```
Bundle hellojavaee.dab.dab not found. Specify the path with --file
```

This makes using the generated file name, particularly using auto completion, difficult to use with `deploy` command. The command should either take both with or without extension, or allow with the extension.

**Comment by thaJeztah** [comment]
Agreed, got bit by this as well; perhaps we should accept both (with and without extension)

@dnephin wdyt?

**Comment by BretFisher** [comment]
Yea seems like not accepting full filename is an anti-patern to all other docker tools. Why even allow without full name? I'm trying to think of another cli tool that does that. Maybe there's some workflow or thought process behind it that I don't get.

**Comment by BretFisher** [comment]
Also should support `-f` to keep with other docker cli's.

**Comment by thaJeztah** [comment]
> Also should support `-f` to keep with other docker cli's.

We decided to drop the `-f` _for now_, because shorthand flags can easily conflict (`--format`/ `-f`, `--force` / `-f`). It's easy to add back though, and I'm not excluding that option; we're just being careful after making mistakes in the past (and _removing_ a flag is way harder than adding it)

**Comment by dnephin** [comment]
Yes, this will be fixed for 1.13

**Comment by yongtang** [comment]
Created a PR #27322 to address this issue.
